scripts/polkadotjs_endpoints_to_preconfigured.py

```diff
@@ -0,0 +1,248 @@
+import json
+import re
+import os
+import sys
+from pathlib import Path
+import requests
+from scripts.utils.metadata_interaction import get_properties
+from scripts.utils.substrate_interface import create_connection_by_url
+from enum import Enum
+
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+
+class Endpoints(Enum):
+    polkadot = 'https://raw.githubusercontent.com/polkadot-js/apps/master/packages/apps-config/src/endpoints/productionRelayPolkadot.ts'
+    kusama = 'https://raw.githubusercontent.com/polkadot-js/apps/master/packages/apps-config/src/endpoints/productionRelayKusama.ts'
+    westend = 'https://raw.githubusercontent.com/polkadot-js/apps/master/packages/apps-config/src/endpoints/testingRelayWestend.ts'
+    rococo = 'https://raw.githubusercontent.com/polkadot-js/apps/master/packages/apps-config/src/endpoints/testingRelayRococo.ts'
+    paseo = 'https://raw.githubusercontent.com/polkadot-js/apps/master/packages/apps-config/src/endpoints/testingRelayPaseo.ts'
+    # for Paseo change array_matches = re.findall(r'=\s*\[(.*?)\]', content, re.DOTALL)
+    singlechains = 'https://raw.githubusercontent.com/polkadot-js/apps/master/packages/apps-config/src/endpoints/production.ts'
+    testnets = "https://raw.githubusercontent.com/polkadot-js/apps/master/packages/apps-config/src/endpoints/testing.ts"
+
+
+CHAINS_FILE_PATH_DEV = Path(os.getenv("DEV_CHAINS_JSON_PATH", 'chains/v20/chains_dev.json'))
+CHAINS_FILE_PATH_PROD = Path(os.getenv("PROD_CHAINS_JSON_PATH", 'chains/v20/chains.json'))
+
+
+def load_json_file(file_path):
+    if file_path.exists():
+        with open(file_path, 'r') as f:
+            return json.load(f)
+    return []
+
+
+def save_json_file(file_path, data):
+    with open(file_path, 'w') as f:
+        json.dump(data, f, indent=4)
+        f.write('\n')
+
+
+def find_objects(array_content):
+    objects = []
+    stack = []
+    start = -1
+
+    for i, char in enumerate(array_content):
+        if char == '{':
+            if not stack:
+                start = i
+            stack.append(char)
+        elif char == '}':
+            if stack and stack[-1] == '{':
+                stack.pop()
+                if not stack:
+                    objects.append(array_content[start:i + 1])
+            else:
+                # Mismatched braces, handle error as needed
+                pass
+
+    return objects
+
+
+def request_data_from_pjs(file):
+    response = requests.get(file)
+    data = response.json()
+    return data
+
+
+def get_ts_file(url, output_path):
+    response = requests.get(url, stream=True)
+    if response.status_code == 200:
+        with open(output_path, "wb") as file:
+            for chunk in response.iter_content(chunk_size=1024):
+                if chunk:
+                    file.write(chunk)
+        print(f"Downloaded file saved as {output_path}")
+    else:
+        print(f"Failed to download file. Status code: {response.status_code}")
```

Comment:
would be nice to have more wider error handling
```suggestion
def get_ts_file(url, output_path):
    try:
        response = requests.get(url, stream=True)
        response.raise_for_status()
        with open(output_path, "wb") as file:
            for chunk in response.iter_content(chunk_size=1024):
                if chunk:
                    file.write(chunk)
        print(f"Downloaded file saved as {output_path}")
    except Exception as e:
        print(f"Failed to download file from {url}. Error: {e}")
```